datumaro/components/media.py

```diff
@@ -42,6 +42,7 @@ def __init__(self,
             data: Union[np.ndarray, Callable[[str], np.ndarray], None] = None,
             *,
             path: Optional[str] = None,
+            ext: Optional[str] = None,
```

Comment:
I don't understand why `Image` needs this parameter. In the original bug report, the issue was that the original extension was lost after resize, but in that case, the full path is available, so it would be sufficient to copy that.

Furthermore, what would the semantics of this parameter be? In `ByteImage` it specifies the encoding of the image data, but for plain `Image` it seems to be meaningless.

Reply:
> In the original bug report, the issue was that the original extension was lost after resize, but in that case, the full path is available, so it would be sufficient to copy that.

If we just store the original path, we'll not write the updated data in https://github.com/openvinotoolkit/datumaro/blob/develop/datumaro/components/media.py#L119-L123, which optimizes the saving time significantly in regular cases. The existence of the `path` serves as an indicator of data modification.

> Furthermore, what would the semantics of this parameter be?

The same as in `ByteImage` - the desired, or the original file extension.

Reply:
OK, I think I understand.